subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -13,26 +13,25 @@
 #define LOG_MODULE_NAME bt_ctlr_llsw_nordic_lll_clock
 #include "common/log.h"
 #include "hal/debug.h"
-
-#define DRV_NAME DT_NORDIC_NRF_CLOCK_0_LABEL "_32K"
-#define K32SRC   CLOCK_CONTROL_NRF_K32SRC
-
-static u8_t is_k32src_stable;
+#include "hal/cpu.h"
 
 void lll_clock_wait(void)
 {
-	if (!is_k32src_stable) {
-		struct device *clk_k32;
+	static bool done;
+
+	if (done) {
+		return;
+	}
+	done = true;
 
-		is_k32src_stable = 1U;
+	struct device *lf_clock = device_get_binding(
+		DT_NORDIC_NRF_CLOCK_0_LABEL "_32K");
 
-		clk_k32 = device_get_binding(DRV_NAME);
-		LL_ASSERT(clk_k32);
+	LL_ASSERT(lf_clock);
 
-		while (clock_control_on(clk_k32, (void *)K32SRC)) {
-			DEBUG_CPU_SLEEP(1);
-			k_cpu_idle();
-			DEBUG_CPU_SLEEP(0);
-		}
+	while (clock_control_on(lf_clock, (void *)CLOCK_CONTROL_NRF_K32SRC)) {
+		DEBUG_CPU_SLEEP(1);
+		cpu_sleep();
```

Comment:
why? `k_cpu_idle()` is the standard Zephyr way of calling `WFI/WFE`

Reply:
This is a direct port of #11021. If you want, I can change it to k_cpu_idle.

Reply:
use k_cpu_idle.